src/services/logger.js

```diff
@@ -50,7 +50,14 @@ function createLogger(level = defaultLogLevel) {
     return this._logLevel;
   };
   logger.setLogLevel = function (logLevel) {
-    this._logLevel = logLevel;
+    if (logLevels.includes(logLevel)) {
+      this._logLevel = logLevel;
+    } else {
+      console.warn(
+        `Attempting to set unrecognized log level: ${logLevel}. Skipping.`
+      );
+      console.warn(`Valid log levels are ${logLevels.join(", ")}`);
+    }
```

Comment:
The goal here is to help prevent users from misconfiguring the logger by rejecting invalid log level sets.